vendor/bat-native-ads/src/bat/ads/internal/ml/data/vector_data.cc

```diff
@@ -0,0 +1,104 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <limits>
+#include <map>
+#include <stdexcept>
+#include <vector>
+
+#include "bat/ads/internal/ml/data/vector_data.h"
+
+namespace ads {
+namespace ml {
+namespace data {
+
+VectorData::VectorData() : Data(DataType::VECTOR_DATA) {}
+
+VectorData::VectorData(const VectorData& vector_data)
+    : Data(DataType::VECTOR_DATA) {
+  dimension_count_ = vector_data.GetDimensionCount();
+  data_ = vector_data.GetRawData();
+}
+
+VectorData& VectorData::operator=(const VectorData& vector_data) {
+  dimension_count_ = vector_data.GetDimensionCount();
+  data_ = vector_data.GetRawData();
+  return *this;
+}
+
+VectorData::~VectorData() = default;
+
+VectorData::VectorData(int dimension_count,
+                       const std::map<unsigned, double>& data)
+    : Data(DataType::VECTOR_DATA) {
+  dimension_count_ = dimension_count;
+  data_.reserve(dimension_count_);
+  for (auto data_it = data.begin(); data_it != data.end(); data_it++) {
```

Comment:
`const`

Reply:
nit: could we change `data_it` to `iter` to match rest of code base

Reply:
I think there is no way of making the map iterator `const` because we increment it at each iteration. Have I misunderstood the comment?

Reply:
ah yes, please ignore that comment :-)

Reply:
Done